developer/reference/articles/tooling/blade-cli/08-extending-blade-cli/05-creating-a-blade-profile.markdown

```diff
@@ -0,0 +1,113 @@
+---
+header-id: creating-a-blade-profile
+---
+
+# Creating a Blade Profile
+
+[TOC levels=1-4]
+
+There are two steps when adding a new Blade profile:
+
+- [Creating a new profile](#creating-a-new-profile)
+- [Setting the profile in Liferay Workspace](#setting-a-profile)
+
+You'll learn how to create a profile first.
+
+## Creating a New Profile
+
+To create a new Blade profile, follow these steps:
+
+1.  [Create a generic OSGi module](/docs/7-2/reference/-/knowledge_base/r/creating-a-project).
+
+2.  To create a new command, create the command and arguments classes extending
+    [`BaseCommand`](https://github.com/liferay/liferay-blade-cli/blob/master/cli/src/main/java/com/liferay/blade/cli/command/BaseCommand.java)
+    and
+    [`BaseArgs`](https://github.com/liferay/liferay-blade-cli/blob/master/cli/src/main/java/com/liferay/blade/cli/command/BaseArgs.java),
+    respectively, as described in the
+    [Creating Custom Commands](/docs/7-2/reference/-/knowledge_base/r/creating-custom-commands-for-blade-cli)
+    article. These classes should reside in the profile module's
+    `src/main/java/PACKAGE_NAME` folder. These classes register your command and
+    arguments to Blade CLI.
+
+3.  To override a default command, create new command and arguments classes
+    similar to what was done
+    [here](/docs/7-2/reference/-/knowledge_base/r/creating-custom-commands-for-blade-cli).
+    Instead of extending the
+    [`BaseCommand`](https://github.com/liferay/liferay-blade-cli/blob/master/cli/src/main/java/com/liferay/blade/cli/command/BaseCommand.java)
+    and
+    [`BaseArgs`](https://github.com/liferay/liferay-blade-cli/blob/master/cli/src/main/java/com/liferay/blade/cli/command/BaseArgs.java),
+    classes, extend the command/arguments classes defined for the command you
+    intend to override. Make sure to also set the `@Parameters` annotation's
+    `commandNames` argument to the command to override.
+
+    For example, if you intend to override the default `deploy` command, your
+    arguments class declaration would look like this:
+
+    ```java
+    @Parameters(commandDescription = "Overridden Deploy Command", commandNames = "deploy")
+    public class OverriddenArgs extends DeployArgs {
+
+    }
+    ```
+
+    You can search for the default command/arguments classes
+    [here](https://github.com/liferay/liferay-blade-cli/tree/master/cli/src/main/java/com/liferay/blade/cli/command).
+
+4.  To associate a command to your new profile, set the
+    [`BladeProfile`](https://github.com/liferay/liferay-blade-cli/blob/master/cli/src/main/java/com/liferay/blade/cli/command/BladeProfile.java)
+    annotation in your command class:
+
+    ```java
+    @BladeProfile("myprofile")
+    public class NewCommand extends BaseCommand<NewArgs> {
+
```

Comment:
Should we add this in here, to give the user a better idea of a brand new command? 
```
public void execute() throws Exception {
}
@Override
public Class<NewArgs> getArgsClass() {
      return NewArgs.class;
}
```

Maybe the `Hello` command example was enough for the reader to understand these methods are necessary in a Command class.